Evaluating Machine Learning Model Performance

collins-a opened the pull request:
Article as index.md
Hero image as hero.jpg

Additional images:
fscore.jpg, mae.jpg, mse.jpg, confusion.png, perecisionrecall.png

### Article Checklist
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

collins-a commented:
Fixes #633

hectorkambow commented:
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

hectorkambow commented:
[Evaluating Machine Learning Model Performance #675.pdf](https://github.com/section-io/engineering-education/files/5584977/Evaluating.Machine.Learning.Model.Performance.675.pdf)
@collins-a 👋  Good morning - just an FYI - upon running your article through our 3rd party plagiarism checker it raised a couple red flags ( a few more than we like to see) nothing too major.
Please go over the PDF and check if any extra sentences need extra links or sources. 
👍  
Thank you

collins-a commented:
Hello @hectorkambow, apologies for that and thank you for the heads up. Let me sort it out asap 😁

collins-a commented:
@hectorkambow hey! I've added a number of links to the flagged sections that were of relevance to the article. I'll be keener next time. Let me know if I've done the correct thing or if there are more changes to be made

lalith1403 commented on articles/evaluating-ml-model-performance/index.md:
```diff
@@ -0,0 +1,157 @@
+[Source](https://towardsdatascience.com/metrics-to-evaluate-your-machine-learning-algorithm-f10ba6e38234)
+
+##### Root Mean Squared Error
+
```
Why is RMSE used? Mention that as well. (Smoother gradients and L2 norm).

lalith1403 commented on articles/evaluating-ml-model-performance/index.md:
```diff
@@ -0,0 +1,157 @@
+
+##### Confusion Matrix
+
+The confusion matrix forms the basis for the other types of classification metrics. It is a matrix that fully describes the performance of the model. A confusion matrix gives an in-depth breakdown of the correct and incorrect classifications of each class.
```
In a confusion matrix, what values are acceptable? Is it alright to have false positives? Is it alright to have true negatives? Which ones should be minimized? What insights does one get from CM? Answer these questions. Moreover, define precision and recall intuitively. In terms of the memory property of the model.

lalith1403 reviewed (changes requested):
A few changes and this should be good to go. A Grammarly check after edits is advisable to ensure minimal rounds of revision.

collins-a commented:
@lalith1403 hey, I've made some additions but I struggled when it came to what is meant by the memory property of the model as well as on the part of normalization. I might need some time for some personal research on normalization and regularization. Thanks for the challenge (and a potential new topic to write on). However, for the sake of the article, I've altered and added what I can at this moment. I've included a link to an article on RMSE to give further information in addition to the paragraph I added. 

Done a Grammarly check as well. The few alerts that remain are basically false positives.

Let me know if these changes are adequate. Thanks again!
